src/lib/Preview.js

```diff
@@ -492,8 +499,14 @@ class Preview extends EventEmitter {
         // Save reference to the currently shown file, if any
         const currentFileId = this.file ? this.file.id : undefined;
 
-        // Use cached file data if available, otherwise create empty file object
-        this.file = this.cache.get(fileId) || { id: fileId };
+        // Check if file ID or well-formed file object was passed in
+        if (typeof fileIdOrFile === 'string') {
+            // Use cached file data if available, otherwise create empty file object
+            this.file = this.cache.get(fileIdOrFile) || { id: fileIdOrFile };
+        } else {
```

Comment:
Make this `else if checkValidFile()` else throw the error. So that `load()` function is guarded.